src/metabase/lib/schema.cljc

```diff
@@ -33,8 +35,19 @@
 (mr/def ::stage.native
   [:map
    [:lib/type [:= :mbql.stage/native]]
-   [:native any?]
-   [:args {:optional true} [:sequential any?]]])
+   ;; the actual native query, depends on the underlying database. Could be a raw SQL string or something like that.
+   ;; Only restriction is that it is non-nil.
+   [:native some?]
+   ;; any parameters that should be passed in along with the query to the underlying query engine, e.g. for JDBC these
+   ;; are the parameters we pass in for a `PreparedStatement` for `?` placeholders. These can be anything, including
+   ;; nil.
+   [:args {:optional true} [:sequential any?]]
+   ;; the Table/Collection/etc. that this query should be executed against; currently only used for MongoDB, where it
+   ;; is required.
+   [:collection {:optional true} ::common/non-blank-string]
```

Comment:
Shouldn't this be part of `:native`? If this is not there, why cannot we restrict `:native` to `:string`? (I can imagine that changing this would imply a big chain of changes that we don't want to do in this PR.)

Reply:
This files comes from https://github.com/metabase/metabase/pull/31789
`:collection` is in the correct place here, I imagine `:query` is unrestricted to be db agnostic.

```clojure
(mc/validate ::lib.schema/query
             (lib.convert/->pMBQL
               {:native   {:query      "[{\"$project\": {\"_id\": \"$_id\"}},
                                        {\"$match\": {\"_id\": {\"$eq\": 1}}},
                                        {\"$group\": {\"_id\": null, \"count\": {\"$sum\": 1}}},
                                        {\"$sort\": {\"_id\": 1}},
                                        {\"$project\": {\"_id\": false, \"count\": true}}]"
                           :collection "venues"}
                :type     :native
                :database 5}))
```